src/view-utilities.js

```diff
@@ -153,29 +154,79 @@ var viewUtilities = function (cy, options) {
       shiftKeyDown = false;
     }, "keyup");
 
-    var p_start_x, p_start_y, p_end_x, p_end_y;
+    var rect_start_pos_x, rect_start_pos_y, rect_end_pos_x, rect_end_pos_y;
 
-    cy.one('tapstart', function( event){ 
+    cy.one('tapstart', function(event){ 
       if( shiftKeyDown == true){
-        p_start_x = event.position.x;
-        p_start_y = event.position.y;
+        rect_start_pos_x = event.position.x;
+        rect_start_pos_y = event.position.y;
+
       }
     });
 
-    cy.one('tapend', function( event){
-      p_end_x = event.position.x;
-      p_end_y = event.position.y;
-      var zoomLevel = Math.min( cy.width()/ ( Math.abs(p_end_x- p_start_x)), cy.height() / Math.abs( p_end_y - p_start_y));
+    cy.one('tapend', function(event){
+      rect_end_pos_x = event.position.x;
+      rect_end_pos_y = event.position.y;
+      //Find top left of rectangle
+      if( rect_start_pos_x > rect_start_pos_x){
+        var temp = rect_start_pos_x;
+        rect_start_pos_x = rect_end_pos_x;
+        rect_end_pos_x = temp;
+      }
+      if( rect_start_pos_y > rect_end_pos_y){
+        var temp = rect_start_pos_y;
+        rect_start_pos_y = rect_end_pos_y;
+        rect_end_pos_y = temp;
+      }
+      //Calculate zoom level
+      var zoomLevel = Math.min( cy.width()/ ( Math.abs(rect_end_pos_x- rect_start_pos_x)), 
+        cy.height() / Math.abs( rect_end_pos_y - rect_start_pos_y));
+
+      //Check whether rectangle intersects with bounding box of the graph
+      //if not abort marquee zoom
+      if((Math.min(rect_start_pos_x, rect_end_pos_x) > cy.elements().boundingBox().x2)
+        ||(Math.max(rect_start_pos_x, rect_end_pos_x) < cy.elements().boundingBox().x1)
+        ||(Math.min(rect_start_pos_y, rect_end_pos_y) > cy.elements().boundingBox().y2)
+        ||(Math.max(rect_start_pos_y, rect_end_pos_y) < cy.elements().boundingBox().y1)){
+        cy.autounselectify(false);
+        cy.elements().unselect();
+        return;        
+      }
+      //if zoom level reaches max abort marquee zoom
+      if(zoomLevel > cy.maxZoom()){
```

Comment:
Since we will have a rectangle whose width and height values are at least 100, I don't think that zoom level will reach to the maxZoom level. This may be an unnecessary check.

Reply:
@hasanbalci this checks whether the selected region is a point( zoom level becomes infinity) for the bug mentioned at comments but as you said I removed it since we extend sides of the box to 100px.